server/common/src/main/java/org/eclipse/lsp/cobol/common/mapping/ExtendedDocument.java

```diff
@@ -208,16 +208,37 @@ public boolean isLineEmptyBetweenColumns(int lineno, int start, int end) {
 
   private Range updateRangeDueToChanges(Range range) {
     if (isDirty()) {
-      range = new Range(updatePositionDueToChanges(range.getStart()), updatePositionDueToChanges(range.getEnd()));
+      if(Objects.equals(range.getStart(), range.getEnd())) {
+        range = new Range(updatePositionDueToChanges(range.getStart()), updatePositionDueToChanges(range.getEnd()));
+      } else {
+        range = new Range(updatePositionDueToChanges(range.getStart()), shiftRight(updatePositionDueToChanges(shiftLeft(range.getEnd()))));
```

Comment:
end `0, 0` is transformed into `0, 1`.

Reply:
it should not, because it's ether 0,0 - 0,0 or incorrect range.

Reply:
`0,0 -> shiftLeft -> 0,0 -> (updatePositionDueToChanges) -> 0,0 -> shiftRight -> 0,1`